tests/queries/0_stateless/02581_share_big_sets_between_multiple_mutations_tasks_long.sql

```diff
@@ -18,12 +18,15 @@ SELECT name FROM system.parts WHERE database=currentDatabase() AND table = '0258
 
 -- Start multiple mutations simultaneously
 SYSTEM STOP MERGES 02581_trips;
-ALTER TABLE 02581_trips UPDATE description='5' WHERE id IN (SELECT (number*10 + 5)::UInt32 FROM numbers(200000000)) SETTINGS mutations_sync=0;
-ALTER TABLE 02581_trips UPDATE description='6' WHERE id IN (SELECT (number*10 + 6)::UInt32 FROM numbers(200000000)) SETTINGS mutations_sync=0;
-ALTER TABLE 02581_trips DELETE WHERE id IN (SELECT (number*10 + 7)::UInt32 FROM numbers(200000000)) SETTINGS mutations_sync=0;
-ALTER TABLE 02581_trips UPDATE description='8' WHERE id IN (SELECT (number*10 + 8)::UInt32 FROM numbers(200000000)) SETTINGS mutations_sync=0;
+ALTER TABLE 02581_trips UPDATE description='5' WHERE id IN (SELECT (number*10 + 5)::UInt32 FROM numbers(10000000)) SETTINGS mutations_sync=0;
+ALTER TABLE 02581_trips UPDATE description='6' WHERE id IN (SELECT (number*10 + 6)::UInt32 FROM numbers(10000000)) SETTINGS mutations_sync=0;
+ALTER TABLE 02581_trips DELETE                 WHERE id IN (SELECT (number*10 + 7)::UInt32 FROM numbers(10000000)) SETTINGS mutations_sync=0;
+ALTER TABLE 02581_trips UPDATE description='8' WHERE id IN (SELECT (number*10 + 8)::UInt32 FROM numbers(10000000)) SETTINGS mutations_sync=0;
 SYSTEM START MERGES 02581_trips;
-DELETE FROM 02581_trips WHERE id IN (SELECT (number*10 + 9)::UInt32 FROM numbers(200000000));
+DELETE FROM 02581_trips                        WHERE id IN (SELECT (number*10 + 9)::UInt32 FROM numbers(10000000));
 SELECT count(), _part from 02581_trips WHERE description = '' GROUP BY _part ORDER BY _part;
 
+SYSTEM FLUSH LOGS;
+SELECT DISTINCT peak_memory_usage < 2000000000 ? 'Ok' : toString(tuple(*)) FROM system.part_log WHERE database = currentDatabase() AND event_date >= yesterday() AND table = '02581_trips' AND event_type = 'MutatePart';
```

Comment:
The test is aimed to check that a big set that is built for `where in (...)` condition is not created for each part, but is shared between mutations tasks for different parts that run in parallel. But AFAIU `peak_memory_usage` column only includes memory consumption by a single thread that is mutating one part.

So it seems that the check query should be different. I will think a bit and try to come up with a proper check

Reply:
I've modified the check to look at `read_rows` from `system.part_log` and compare it to the size of big set for each mutated part. It should be `>= set size` for the part that builds the set and `<= set size` for those parts that used shared set instead of building it.

Reply:
The approach with `read_rows` didn't work out with old analyzer.
I changed the check to count log messages like `Created Set with 10000000 entries from 10000000 rows in 0.388989187 sec.` and `Got set from cache in 0.388930505 sec.`